packages/flutter/lib/src/material/adaptive_text_selection_toolbar.dart

```diff
@@ -0,0 +1,312 @@
+/// * [TextField.contextMenuBuilder]
+/// * [CupertinoTextField.contextMenuBuilder]
+/// * [SelectionArea.contextMenuBuilder]
+/// * [SelectableText.contextMenuBuilder]
+///
+/// See also:
+///
+/// * [EditableTextContextMenuButtonItemsBuilder], which builds the
+///   [ContextMenuButtonItem]s.
+/// * [TextSelectionToolbarButtonsBuilder], which builds the button Widgets
+///   given [ContextMenuButtonItem]s.
+/// * [CupertinoAdaptiveTextSelectionToolbar], which does the same thing as this
+///   widget but only for Cupertino context menus.
+/// * [TextSelectionToolbar], the default toolbar for Android.
+/// * [DesktopTextSelectionToolbar], the default toolbar for desktop platforms
+///    other than MacOS.
+/// * [CupertinoTextSelectionToolbar], the default toolbar for iOS.
+/// * [CupertinoDesktopTextSelectionToolbar], the default toolbar for MacOS.
+class AdaptiveTextSelectionToolbar extends StatelessWidget {
+  /// Create an instance of [AdaptiveTextSelectionToolbar] with the
+  /// given [children].
+  ///
+  /// {@template flutter.material.AdaptiveTextSelectionToolbar.constructor.editableText}
+  /// To adaptively generate the default buttons for [EditableText] for the
+  /// current platform, use [AdaptiveTextSelectionToolbar.editableText].
+  /// {@endtemplate}
+  ///
+  /// {@template flutter.material.AdaptiveTextSelectionToolbar.constructor.selectableRegion}
+  /// To adaptively generate the default buttons for [SelectableRegion] for the
+  /// current platform, use [AdaptiveTextSelectionToolbar.selectableRegion].
+  /// {@endtemplate}
+  ///
+  /// {@template flutter.material.AdaptiveTextSelectionToolbar.constructor.buttonItems}
+  /// To specify the button labels and callbacks but still adaptively generate
+  /// the look of the buttons based on the current platform, use
+  /// [AdaptiveTextSelectionToolbar.buttonItems].
+  /// {@endtemplate}
+  const AdaptiveTextSelectionToolbar({
+    super.key,
+    required this.children,
+    required this.primaryAnchor,
+    this.secondaryAnchor,
+  }) : buttonItems = null,
+       editableTextState = null,
+       selectableRegionState = null;
+
+  /// Create an instance of [AdaptiveTextSelectionToolbar] and
+  /// adaptively generate the buttons based on the current platform and the
+  /// given [editableTextState].
+  ///
+  /// {@template flutter.material.AdaptiveTextSelectionToolbar}
+  /// To specify the [children] widgets directly, use the main constructor
+  /// [AdaptiveTextSelectionToolbar.AdaptiveTextSelectionToolbar].
+  /// {@endtemplate}
+  ///
+  /// {@macro flutter.material.AdaptiveTextSelectionToolbar.constructor.selectableRegion}
+  ///
+  /// {@macro flutter.material.AdaptiveTextSelectionToolbar.constructor.buttonItems}
+  ///
+  /// See also:
+  ///
+  ///  * [EditableTextContextMenuButtonItemsBuilder], which builds the default
+  ///    [ContextMenuButtonItem]s for [EditableText] on the platform.
+  const AdaptiveTextSelectionToolbar.editableText({
```

Comment:
Why grouping them in AdaptiveTextSelectionToolbar. I looked at the build, they are their own ContextMenuButtonItemsBuilder anyway

Reply:
The idea was just user friendliness, but now I'm thinking you might be right and they should just be separate widgets.  I've split them out, let me know what you think.